src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/IfError.cs

```diff
@@ -37,25 +38,56 @@ public IfErrorFunction()
                     new TypedName(ErrorType.ReifiedErrorTable(), new DName("AllErrors")),
                     new TypedName(DType.ObjNull, new DName("ErrorResult"))),
                 appliesToArgument: i => i > 0 && (i % 2 == 1));
+
+            // IfError(value1, fallback1, value2, fallback2, ..., valueN, [fallbackN], ...)
+            SignatureConstraint = new SignatureConstraint(omitStartIndex: 4, repeatSpan: 2, endNonRepeatCount: 0, repeatTopLength: 8);
         }
 
         public override IEnumerable<TexlStrings.StringGetter[]> GetSignatures()
         {
             yield return new[] { TexlStrings.IfErrorArg1, TexlStrings.IfErrorArg2 };
-            yield return new[] { TexlStrings.IfErrorArg1, TexlStrings.IfErrorArg2, TexlStrings.IfErrorArg2 };
-            yield return new[] { TexlStrings.IfErrorArg1, TexlStrings.IfErrorArg2, TexlStrings.IfErrorArg2, TexlStrings.IfErrorArg2 };
         }
 
         public override IEnumerable<TexlStrings.StringGetter[]> GetSignatures(int arity)
         {
             if (arity > 2)
             {
-                return GetGenericSignatures(arity, TexlStrings.IfErrorArg2);
+                return GetOverloadsIfError(arity);
             }
 
             return base.GetSignatures(arity);
         }
 
+        // Gets the overloads for the IfError function for the specified arity.
+        // If is special because it doesn't have a small number of overloads
+        // since its max arity is int.MaxSize.
```

Comment:
hmm, this isn't necessarily special. We have a bunch of variadic functions that behave this way, e.g. AddColumns, Sum, Table, If/Switch, ...

Maybe this can be standardized/parameterized better?

Reply:
not something you need to address in this PR obviously,